src/app/inventory/store/crafted.ts

```diff
@@ -9,6 +14,21 @@ export const craftedSocketCategoryHash = 3583996951;
 /** the socket category containing the Mementos */
 export const mementoSocketCategoryHash = 3201856887;
 
+/**
+ * Generate a table from item name to the record for their crafting pattern.
+ */
+export const itemNameToCraftingPatternRecordHash = _.once((defs: D2ManifestDefinitions) => {
+  const recordHashesByName: { [itemName: string]: number } = {};
+  if (defs) {
+    for (const record of Object.values(defs.Record.getAll())) {
+      if (record.completionInfo?.toastStyle === DestinyRecordToastStyle.CraftingRecipeUnlocked) {
+        recordHashesByName[record.displayProperties.name] = record.hash;
+      }
+    }
+  }
+  return recordHashesByName;
+});
```

Comment:
* Matching by name is really fuzzy, and I can't tell if that's good or bad. Example: Bungie is creating craftable versions of the old Menagerie weapons with the same names but slightly different perk pools. This would treat the pattern record as applying to old sunset copies too even though they're different hashes and have different perk pools.
  * There's a decent case to be made that this is intended behavior, but it'd be good to acknowledge this. Otherwise we could match a record to a crafting recipe item first and then associate with the recipe's outputItemHash; that'd always match by exact hash.
* Should this be in D2AI?
* Unrelated items can have the same name:
![grafik](https://user-images.githubusercontent.com/14299449/170460745-ab81401f-7472-4fc8-9fca-1da1622343ef.png)

Reply:
My preference in general is to generate these tables at runtime instead of D2AI where possible, to avoid needing an update cycle to react to changes - in many cases the D2AI logic isn't any more sophisticated than this name matching. I'll look into how we can validate that they're actually patterns though - at the very least a limit to only matching weapons is worthwhile. I'm not super worried about reissues since the old menagerie weapons are sunset, right?

Reply:
The old Menagerie weapons are indeed sunset (with the caveat that we don't know Bungie's future plans for the next set of reissued craftable weapons) and I can't say that matching them would be wrong -- just wanted to quickly confirm that this is fine.

Reply:
I updated the validation I'm doing - now those items shouldn't match, but the downside is that some things that should match (Adept Vow weapons) can't match now.